### PR TITLE
feat: Version to be dynamic from file

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Made version to be dynamic and read from a file
- MINOR: Added a gitignore entry for pdm related files

## Why

- Makes it consistent with the SDK
- Allows for retrieving the package version at runtime, this can help with logging and gives visibility on the version of packages being used

## How

...

## Relevant Docs

- [PDM dynamic version](https://backend.pdm-project.org/metadata/#dynamic-project-version)
- PDM also supports reading version from the SCM tag however I haven't dived deeper into this yet and it can be considered if necessary later

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

-

## Notes on Testing
- Tried to publishing to a local PyPI server and was able to read the version at runtime
```
# Run server
pypi-server run -v -a . -P . -p 2000 --overwrite packages

# Publish to local server
pdm publish --repository http://localhost:2000/

# Pull from server
pip install --index-url http://localhost:2000 'unstract-adapters==0.4.0'
```

## Screenshots
- PyPI server logs
image
- Package published
image
- Package pulled and version checked
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by hari-kuriakose — APPROVED** [review]
@chandrasekharan-zipstack LGTM

**Review by jaseemjaskp — APPROVED** [review]

File: src/unstract/adapters/__init__.py [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 import logging
 from logging import NullHandler
 from typing import Any
 
 logging.getLogger(__name__).addHandler(NullHandler())
 
 AdapterDict = dict[str, dict[str, Any]]
+
+
+def get_adapter_version() -> str:
```
Comment (gaya3-vijayakumar) [inline src/unstract/adapters/__init__.py:12]: Just one question @chandrasekharan-zipstack How does the system know that it needs to call this particular function as we are not referring to it anywhere? We only mention the filename in __init__.py

Comment (gaya3-vijayakumar) [reply src/unstract/adapters/__init__.py:12]: Or this is just an enabler I guess? Just noticed line 1 __version__ = "0.4.0" Guess that is the one doing the trick?

**Review by gaya3-vijayakumar — APPROVED** [review]
Looks fine

File: src/unstract/adapters/__init__.py (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/adapters/__init__.py:12]: Yeah @gaya3-zipstack - its handled by [PDM](https://backend.pdm-project.org/metadata/#dynamic-project-version)

**Pull request merged**